### ISSUE TITLE
Request/Question: It is possible to make an improvement to support BulkRequests to routes? 

### ISSUE THREAD

**Issue by marvel16** [edjCase/JsonRpc#96]
e.g. i have multiple rpccontrollers 
api/controller1
                     /method1
                     /method2
api/controller2
                    /method3
                    /method4
api/controller3
                    /method5

Would be really nice to have router/client interact with different endpoints.

so I'd send requests to method1, method3, method5 in the same bulk.

**Comment by Gekctek** [comment]
Based on the json rpc 2 spec there seems to be only a route (url path) and the method name. You could have the same controllers be accessable by the same path with configuration but what are you looking for exactly

**Comment by marvel16** [comment]
Ok got it, so the solution here would be making multiple controllers available under the same route?

The thing about our implementation is that we decided to inject our rpcclient wrapper with methods that leverage CallerMemberName attribute.

This allows us to inject our rpcclient wrapper into any class that implements interface on client side and whenever you invoke rpcclient the request is automatically determining the method and sends it to server into controller that implements this same interface.

![image](https://user-images.githubusercontent.com/5057793/106849231-02fd3c00-66bb-11eb-945d-fb6d74549382.png)

The issue we want to solve for performance optipisation/network lag where multiple requests are sent sequentially in multiple places. As requests are independent we could send them in parallel or use this send bulk feature inside single httpcall that your library provides.

FYI 
we are still on .net472, but using .netcore api style because some nugets are holding us from upgrade to higher .net standard20 packages because of missing method exceptions on some Microsoft.Extensions.Configuration dependencies like Primitives...

So our version of your package that lets API not crash is  2.X.X latest

**Comment by Gekctek** [comment]
Ya something like:

```cs
app.UseJsonRpc(options =>
    {
    	options
    		.AddControllerWithCustomPath<Controller1>("/")
    		.AddControllerWithCustomPath<Controller2>("/");
    });
```
might work or putting them all the methods in the same controller

I dont think there is a lot of wiggle room because the JsonRpc2 spec can't be changed

let me know if i can help with anything else

**Comment by marvel16** [comment]
@Gekctek 
Thanks for clarification, I think we'll come with attributes to match the route among controllers that logically contain different logic but might be invoked in bulk.
